drilldown/utils.py

```diff
@@ -0,0 +1,14 @@
+
+def ordered_unique_fields(query_string):
+    '''
+        Returns an ordered list of unique field names
+        based on the query string passed in
+    '''
+    all_request_fields = [x.split('=')[0] for x in query_string.split('&')]
+    unique_fields = set(all_request_fields)
+    request_fields = []
+    for field in all_request_fields:
+        if field in unique_fields:
+            request_fields.append(field)
+            unique_fields.remove(field)
```

Comment:
This looks like it would ignore more than one setting for the same parameter.  (e.g. "?a=1&a=2") which we expect to come back as ['1', '2'] from request.GET.

Reply:
Yes, I thought that as well, but then I noticed that this section of code is only interested in the query keys. In the view, we then use those keys to get both values from request.GET.

Reply:
Oh right.  Nevermind.

Reply:
But since we both thought that, maybe a comment might be in order to clarify :)

Reply:
I would probably write this more like

```
unique_fields = []
for field in all_request_fields:
    if field not in unique_fields:
        unique_fields.append(field)
return unique_fields
```

which seems easier to understand to me, I guess because it's just adding things rather than both adding and removing things as it goes along, and it's the way I've always seen this done.  I think the code as written is correct, but I had to think about it a bit. I think I'd vote for a little simpler way of doing it, to make things easier for future maintainers.